src/lib/utils.js

```diff
@@ -0,0 +1,33 @@
+import { PixelRatio, Platform } from 'react-native'
+
+export const applyImgixParameters = (source, layout, imgixProps = {}) => {
+  if (!layout) {
+    return null
+  }
+
+  const uri = typeof source === 'object' ? source.uri : source
+
+  if (typeof uri !== 'string' || !uri.includes('imgix.net')) {
+    return source
+  }
+
+  const path = uri.split(/[?#]/)[0]
+  const params = {
+    ...(layout.width && { w: layout.width }),
+    ...(layout.height && { h: layout.height }),
+    ...(typeof imgixProps === 'object' && { ...imgixProps }),
+    dpr: PixelRatio.get(),
+    ...(Platform.OS === 'web' && { auto: 'format,compress' }),
+    ...(Platform.OS !== 'web' && { auto: 'compress' }),
+  }
```

Comment:
```
const params = {
   w: layout.width,
   h: layout.height,
   ...(typeof imgixProps === 'object' && { ...imgixProps }),
   dpr: PixelRatio.get(),
   auto: Platform.OS === 'web' ? 'format,compress' : 'compress',
  }
```

I find this a lot more legible and mostly equivalent for what we are trying to do here, no?
(also spreading can be real slow depending on the browser or js engine.